src/asmcnc/keyboard/custom_keyboard.py

```diff
@@ -154,13 +138,33 @@ def set_keyboard_background(self):
             self.margin_hint = [.05, .06, .05, .06]  # Default margin
             self.background = "atlas://data/images/defaulttheme/vkeyboard_background"
 
+        if self.layout == self.numeric_layout:
+            self.width = Window.width / 3
+        else:
+            self.width = Window.width
+
+        # Make sure keyboard never goes off-screen and becomes unusable/unreachable
+        if self.pos[0] + self.width > Window.width:
+            self.pos = (Window.width - self.width, self.pos[1])
+        if self.pos[1] < 0:
+            self.pos = (self.pos[0], 0)
+        if self.pos[0] < 0:
+            self.pos = (0, self.pos[1])
+        if self.pos[1] + self.height > Window.height:
+            self.pos = (self.pos[0], Window.height - self.height)
```

Comment:
Same as above

Reply:
This breaks functionality when the keyboard leaves the screen on the top and right.

Reply:
But the value should be the same? Confused what you mean.